proto/src/main.cpp

```diff
@@ -1,73 +1,89 @@
 #include "Particle.h"
-#include "Arduino.h"
 #include "JsonMaker.h"
-#include "Sensor_RSSI.h"
-#include "Sensor_ECU.h"
-
-#define BLINK_INTERVAL_OFF 1800
-#define BLINK_INTERVAL_ON 200
-#define PUBLISH_INTERVAL 10000
+#include "SensorEcu.h"
+#include "SensorGps.h"
+#include "SensorThermo.h"
 
+SYSTEM_MODE(AUTOMATIC);
 SYSTEM_THREAD(ENABLED);
 
-void onSerialData();
-JsonMaker json_maker;
-Sensor_RSSI rssi;
-Sensor_ECU ecu(&Serial1);
-
-
-uint32_t last_blink = 0; // Time of last blink
-uint32_t last_publish = 0; // Time of last publish
-boolean led_state = LOW;
+// Disable publish when not necessary during development to reduce data usage
+#define PUBLISH_ENABLED             1
```

Comment:
I like this a lot. One suggestion would be to also wrap the `Serial.print` statements into a macro that's enabled/disabled based on this flag. For example, something like [this](https://stackoverflow.com/questions/31860255/console-output-only-in-debug-mode-c).

Reply:
Done, though added a new flag for this as we will want to output debug messages over serial AND have publish enabled during testing also